contracts/periphery/ShieldedYieldRouter.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.28;
+
+import {IERC20, IERC4626} from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import {IShielder} from "./IShielder.sol";
+
+contract ShieldedYieldRouter {
+    constructor() {}
+
+    /// @notice Mint exactly `amount` shares and deposit them in the `shielder`.
+    /// @dev    The `msg.sender` must approve sufficient assets to this contract.
+    function mintAndShieldWithNewAccount(
+        address shielder,
+        bytes3 expectedContractVersion,
+        address vault,
+        uint256 amount, // shares amount
```

Comment:
I think it would be nice to have documentation for `amount` as natspec's `@param`, since it's unusual for input `amount` to be denoted in shares.